src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.BlockAnalyzer.cs

```diff
@@ -366,7 +366,8 @@ private void AnalyzeOperationBlockEnd(OperationBlockAnalysisContext context)
                         out ImmutableDictionary<string, string> properties)
                     {
                         properties = null;
-                        if (_options.UnusedValueAssignmentSeverity == ReportDiagnostic.Suppress)
+                        if (_options.UnusedValueAssignmentSeverity == ReportDiagnostic.Suppress ||
+                            symbol.GetSymbolType().IsErrorType())
```

Comment:
should we maybe also fail if the symbol type is null?

Reply:
Hmm, I am nit sure that is possible for locals/parameters.

Reply:
Yeah, i'm not sure either...  at one point i tried to convince myself of these sorts of invariants, but ended up finding that it looked like in some (extremely esoteric mind you) cases, you might end up with null.  I sent out mails if that would be considered buggy behavior or not, but i don't remember the resolution.

I'm ok keeping the code as is mind you.  It was just a thought :)

Reply:
> so, I don't see any issue it marking local name as unused.

I would prefer we not mark things when there are obvious errors in the code. 

The intuition here is that clearly the code isn't complete, so overly aggressively telling users to remove things is more annoyance than value.  This is why, for example, i believe we shouldn't offer this feature with syntactic errors as well (and i think that's how the feature works).

Another way to think about is this:  the compiler learned over tiem that overly aggressively giving people errors (or cascading errors) isn't a great experience.  So once an initial error is reported on something, they attempt to suppress (with varying levels of success) further errors.  Depending on the type of change we're offering, we often want to do the same.  

Note that this is dependent on the type of change being offered.  If we are helping them generate/fix code, then we really want to be resilient to errors.  However, if we are removing/destroying code, we want to be conservative and wait until we have higher confidence we know what's going on.

Reply:
if we truly believe that, we should show this kind of diagnostics when we know that users are done with changing code, we should make this build only diagnostics like unused field. otherwise, it is just random point we decide based on our implementation.

even after adding using, it still shows local as unused which I will use soon since I am not done yet.. which someone can say it is noise.

anyway, it wasn't a distraction for me either way. so doesn't matter for me, but if we truly care about people who get distracted easily, sounds like there should be an option to make it build only errors.